bottlecap/src/secrets/decrypt.rs

```diff
@@ -43,23 +56,75 @@ pub fn resolve_secrets(config: Config) -> Result<Config> {
     }
 }
 
+struct RequestArgs<'a> {
+    service: String,
+    body: &'a Value,
+    time: DateTime<Utc>,
+    x_amz_target: String,
+}
+
 struct AwsConfig {
     region: String,
     aws_access_key_id: String,
     aws_secret_access_key: String,
     aws_session_token: String,
+    function_name: String,
 }
 
-fn manual_decrypt(secret_arn: String, aws_config: AwsConfig) -> Result<String> {
+fn decrypt_aws_kms(client: Client, kms_key: String, aws_config: AwsConfig) -> Result<String> {
```

Comment:
If we are already returning a result, shouldn't we also remove all the expects in the code and bubble up the possible errors?

Reply:
That sounds right, thanks!
I used a box dyn error for trying to catch different error types. Let me know if there is a better way to do it!